micrometer-core/src/main/java/io/micrometer/core/instrument/step/StepMeter.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.step;
+
+/**
+ * Internal. Intentionally package-private.
+ */
+interface StepMeter {
```

Comment:
This is internal (also package-private) so it is not the most important thing to bikeshed, but I'm beginning to wonder if a more specific name would be better. `ManualRolloverCapableStepMeter` is long but descriptive.